### PR TITLE
Preserve count option

### PR DESCRIPTION
**Pull request by CrAsH1101**
Some rails validation messages are broken after applying PR #480
More info in comment of #480

### CODE REVIEW

File: lib/i18n/backend/base.rb [hunk 1]
```diff
@@ -163,7 +163,7 @@ def resolve(locale, object, subject, options = EMPTY_HASH)
         #   not standard with regards to the CLDR pluralization rules.
         # Other backends can implement more flexible or complex pluralization rules.
         def pluralize(locale, entry, count)
-          return entry unless entry.is_a?(Hash) && count
+          return entry unless entry.is_a?(Hash) && count && entry.values.select{|v| v.is_a?(Hash)}.size == 0
```
Comment (radar) [inline lib/i18n/backend/base.rb]: Perhaps:

```
entry.values.none? { |v| v.is_a?(Hash) }
```

Comment (CrAsH1101) [reply lib/i18n/backend/base.rb]: Sure

**Comment by radar** [comment]
Could you please add a test to this PR to ensure that the behaviour doesn't get re-broken later on?

**Comment by CrAsH1101** [comment]
Which test file would be most suitable for this in your opinion?

**Comment by radar** [comment]
I figured `fallbacks_test.rb` was the best place for it, since it had lots of other tests for defaults. I've put a test there for your changes now and have verified that your patch does indeed fix the issue.

Thanks again! I'll do a release a little later on today (within an hour or two). Sorry for the trouble caused by #480 and thanks for your patience!

**Comment by caifara** [comment]
Hit this with an semi-obscure message deep in ActiveAdmin (took me a while to get here at least). A release will help more people like me. Thanks for all the hard work!

File: lib/i18n/backend/base.rb (hunk 1 above)
Comment (movermeyer) [reply lib/i18n/backend/base.rb]: To help with archeology:

This implicitly changed the behaviour of pluralization contexts with nested hashes:

```yaml
en:
  stars:
    one: "%{count} star"
    other: "%{count} stars"
    special:
      one: "%{count} special star"
      other: "%{count} special stars"
```

Before:

```ruby
> I18n.t("stars", count: 1)
"1 star"
```

After:

```ruby
> I18n.t("stars", count: 1)
{:one=>"one", :other=>"other", :special=>{:one=>"one special star", :other=>"other special star"}}
```

Which was reported as https://github.com/ruby-i18n/i18n/issues/510, and was reverted in the case of `Backend::Pluralization` in https://github.com/ruby-i18n/i18n/commit/1b5e34553003ca3b42b842769e86c98d5e3b71d4, but not `Backend::Base`. The `Backend::Base` was reported as https://github.com/ruby-i18n/i18n/issues/514.

**Pull request merged**